### PR TITLE
[CDAP-15787] Updated ErrorCollector for new validation API

### PR DESCRIPTION
**Pull request by dli357**
Updated ErrorCollector.java for the new validation API.

### CODE REVIEW

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java [hunk 1]
```diff
@@ -40,28 +39,37 @@
 public class ErrorCollector extends ErrorTransform<StructuredRecord, StructuredRecord> {
   private final Config config;
 
+  private static final String MESSAGE_FIELD = "messageField";
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: put these in the config class

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java [hunk 2]
```diff
@@ -40,28 +39,37 @@
...
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (config.messageField != null && inputSchema.getField(config.messageField) != null) {
-        throw new InvalidConfigPropertyException(String.format(
-          "Input schema already contains message field '%s'. Please set message field to a different value.",
-          config.messageField), "messageField");
+        collector.addFailure(
+            String.format("Input schema already contains message field '%s'.", config.messageField),
+            "Please set message field to a different value.")
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: as we're going through these messages, we've stopped putting 'Please', as it's weird to have it in some places but not others, and sometimes it doesn't make sense.

Set the message field to a different value

Comment (albertshau) [reply core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: similar comments below

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java [hunk 3]
```diff
@@ -40,28 +39,37 @@
...
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (config.messageField != null && inputSchema.getField(config.messageField) != null) {
-        throw new InvalidConfigPropertyException(String.format(
-          "Input schema already contains message field '%s'. Please set message field to a different value.",
-          config.messageField), "messageField");
+        collector.addFailure(
+            String.format("Input schema already contains message field '%s'.", config.messageField),
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: already contains field '%s'.

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java (hunk 1 above)
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java (hunk 2 above)
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java (hunk 3 above)
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: Shouldn't we consider leaving in "Input schema" to be more clear about what already contains that field?

Comment (albertshau) [reply core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: yeah, was lazy and didn't type the whole thing:

Input schema already contains field '%s'.

**Review by albertshau — APPROVED** [review]
lgtm

File: core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java (hunk 3 above)
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java]: Fixed

**Pull request merged**